video/out/mac/gl_layer.swift

```diff
@@ -109,6 +109,13 @@ class GLLayer: CAOpenGLLayer {
         autoresizingMask = [.layerWidthSizable, .layerHeightSizable]
         backgroundColor = NSColor.black.cgColor
 
+        // Experimental #HDR on #macOS        
+        // This must be enabled to reproduce HDR content 
+        if (ccb.libmpv.macOpts.macos_hdr_csp != -1)
+        {
+            wantsExtendedDynamicRangeContent = true
```

Comment:
scattering this option to 3 different places (view, layer, cocoa_cb_common) makes it a bit hard to maintain i believe. 

it might be good idea to set up the whole thing in this function https://github.com/mpv-player/mpv/blob/master/video/out/mac/common.swift#L129-L141 there you have the view and the layer.
you would not need to check the option several times. though you need to type cast the CALayer to a CAOpeGLLayer in an "if let", for new layers we need to add another type casting (MetalLayer in the future)